src/main/java/org/terasology/changingBlocks/conditional/ConditionalBlocksSystem.java

```diff
@@ -0,0 +1,227 @@
+                blockTriggers.add(trigger);
+            }
+        } else {
+            if (!entityTriggers.contains(trigger)) {
+                log.info("Adding trigger for " + trigger);
+                entityTriggers.add(trigger);
+            }
+        }
+        List<EntityRef> triggerTaggers = triggerCollections.computeIfAbsent(trigger, k -> new ArrayList<>());
+        triggerTaggers.add(triggerable);
+    }
+
+    @ReceiveEvent(components = {ChangeBlockBlockDirectedComponent.class, LocationComponent.class, BlockComponent.class})
+    public void onSpawnBlockDirected(OnAddedComponent event, EntityRef entity) {
+        ChangeBlockBlockDirectedComponent changingBlocks = entity.getComponent(ChangeBlockBlockDirectedComponent.class);
+        List<BlockCondition.BlockDirected> listOfChanges = changingBlocks.changes;
+        for (BlockCondition.BlockDirected bc : listOfChanges) {
+            registerTrigger(bc.triggerBlock, entity, true);
+        }
+    }
+
+    @ReceiveEvent(components = {ChangeBlockBlockNearbyComponent.class, LocationComponent.class, BlockComponent.class})
+    public void onSpawnBlockNearby(OnAddedComponent event, EntityRef entity) {
+        ChangeBlockBlockNearbyComponent changingBlocks = entity.getComponent(ChangeBlockBlockNearbyComponent.class);
+        List<BlockCondition.BlockNearby> listOfChanges = changingBlocks.changes;
+        for (BlockCondition.BlockNearby bc : listOfChanges) {
+            registerTrigger(bc.triggerBlock, entity, true);
+        }
+    }
+
+    @ReceiveEvent(components = {ChangeBlockEntityDirectedComponent.class, LocationComponent.class, BlockComponent.class})
+    public void onSpawnEntityDirected(OnAddedComponent event, EntityRef entity) {
+        ChangeBlockEntityDirectedComponent changingBlocks = entity.getComponent(ChangeBlockEntityDirectedComponent.class);
+        List<BlockCondition.EntityDirected> listOfChanges = changingBlocks.changes;
+        for (BlockCondition.EntityDirected ec : listOfChanges) {
+            registerTrigger(ec.triggerEntity, entity, false);
+        }
+    }
+
+    @ReceiveEvent(components = {ChangeBlockEntityNearbyComponent.class, LocationComponent.class, BlockComponent.class})
+    public void onSpawnEntityNearby(OnAddedComponent event, EntityRef entity) {
+        ChangeBlockEntityNearbyComponent changingBlocks = entity.getComponent(ChangeBlockEntityNearbyComponent.class);
+        List<BlockCondition.EntityNearby> listOfChanges = changingBlocks.changes;
+        for (BlockCondition.EntityNearby ec : listOfChanges) {
+            registerTrigger(ec.triggerEntity, entity, false);
+        }
+    }
+
+    @ReceiveEvent(components = {ConditionalBlockChangeComponent.class, LocationComponent.class, BlockComponent.class})
+    public void onRemoving(BeforeRemoveComponent event, EntityRef entity) {
+        for (List<EntityRef> refs : triggerCollections.values())
+        {
+            refs.remove(entity);
```

Comment:
Does this properly update the map? I think `Map.replaceAll` would be a better option here:

```java
triggerCollections.replaceAll( 
    (trigger, refs) -> refs.remove(entity)
)
```